src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp

```diff
@@ -474,31 +474,46 @@ bool SlaterDetBuilder::putDeterminant(xmlNodePtr cur, int spin_group)
     return true;
 
   DiracDeterminantBase* adet = 0;
-  {
+
 #if defined(QMC_CUDA)
-    adet = new DiracDeterminantCUDA(psi, firstIndex);
+  adet = new DiracDeterminantCUDA(psi, firstIndex);
```

Comment:
This deserves a \todo comment (make this logic readable).  minimum verbosity is not a goal for this sort of mess, ideally the design shouldn't require it.

Reply:
Added a TODO.